lib/Sema/TypeCheckNameLookup.cpp

```diff
@@ -434,18 +434,26 @@ static unsigned getCallEditDistance(DeclName argName, DeclName paramName,
   // TODO: maybe ignore certain kinds of missing / present labels for the
   //   first argument label?
   // TODO: word-based rather than character-based?
-  StringRef argBase = argName.getBaseName().str();
-  StringRef paramBase = paramName.getBaseName().str();
-
-  unsigned distance = argBase.edit_distance(paramBase, maxEditDistance);
-
-  // Bound the distance to UnreasonableCallEditDistance.
-  if (distance >= maxEditDistance ||
-      distance > (paramBase.size() + 2) / 3) {
+  if (argName.getBaseName().getKind() != paramName.getBaseName().getKind()) {
     return UnreasonableCallEditDistance;
   }
+  if (argName.getBaseName().getKind() == DeclBaseName::Kind::Normal) {
```

Comment:
Can you make this an early return? If the two kinds are the same and they aren't a "Normal" kind, we can just return 0.